internal/releases/repo_github.go

```diff
@@ -41,41 +41,95 @@ func (r *githubRepository) Process() error {
 	ctx := context.Background()
 	opts := &github.ListOptions{PerPage: 3}
 
+	// Check if the repository is archived
+	repoObject, _, err := client.Repositories.Get(ctx, r.org, r.info.Name)
+	if err != nil {
+		return fmt.Errorf(
+			"error while checking archived status for repo: %s/%s/%s: %v", r.org, r.team, r.info.Name, err,
+		)
+	}
+	r.info.IsArchived = repoObject.GetArchived()
+
+	// Get the releases from the repo
 	releases, _, err := client.Repositories.ListReleases(ctx, r.org, r.info.Name, opts)
 	if err != nil {
 		return fmt.Errorf(
 			"error listing releases for repo: %s/%s/%s: %v", r.org, r.team, r.info.Name, err,
 		)
-	} else if len(releases) == 0 {
-		return nil
-	}
+	} else if len(releases) != 0 { // If there are releases, add them
+		// Iterate over the releases in the Github repo
+		for _, rel := range releases {
+			r.info.Releases = append(r.info.Releases, NewRelease(
+				rel.GetID(),
+				rel.GetTagName(),
+				rel.CreatedAt.Time,
+				rel.GetName(),
+				rel.GetBody(),
+				rel.GetHTMLURL(),
+				fmt.Sprintf("%s/compare/%s...%s", r.info.Url, rel.GetTagName(), r.info.DefaultBranch),
+			))
+		}
+
+		// Add the commit delta between last release and default branch
+		comparison, _, err := client.Repositories.CompareCommits(
+			ctx, r.org, r.info.Name, r.info.Releases[0].Version, r.info.DefaultBranch, opts,
+		)
+
+		if err != nil {
+			return fmt.Errorf(
+				"error getting commit comparison for release %s in %s/%s/%s",
+				r.info.Releases[0].Version, r.org, r.team, r.info.Name,
+			)
+		}
 
-	// Iterate over the releases in the Github repo
-	for _, rel := range releases {
-		r.info.Releases = append(r.info.Releases, NewRelease(
-			rel.GetID(),
-			rel.GetTagName(),
-			rel.CreatedAt.Time,
-			rel.GetName(),
-			rel.GetBody(),
-			rel.GetHTMLURL(),
-			fmt.Sprintf("%s/compare/%s...%s", r.info.Url, rel.GetTagName(), r.info.DefaultBranch),
-		))
+		r.info.NewCommits = *comparison.TotalCommits
+	} else {
+		// If there are no releases, get the latest commit instead
+		commits, _, err := client.Repositories.ListCommits(ctx, r.org, r.info.Name, nil)
+		// If there is at least one commit, add it as a release
+		if err == nil {
+			com := commits[0]
+			r.info.Releases = append(r.info.Releases, NewRelease(
```

Comment:
I think adding this information into `r.info.Releases` is going to cause confusion down the line - I think we should maybe add an `r.info.Commits` field.

In the case where there *are* releases, it'll never get populated, and vice versa - thoughts?

Reply:
It does look cleaner indeed, adding it!